yapf/yapflib/format_decision_state.py

```diff
@@ -934,16 +934,19 @@ def _GetNewlineColumn(self):
     previous = current.previous_token
     top_of_stack = self.stack[-1]
 
+    cont_aligned_indent = self._IndentWithContinuationAlignStyle(
+        top_of_stack.indent)
+
     if isinstance(current.spaces_required_before, list):
-      # Don't set the value here, as we need to look at the lines near
-      # this one to determine the actual horizontal alignment value.
-      return 0
+      # only when the commet is not inside an object(list, dictionary,
+      # function call)logical line that is in many output lines
```

Comment:
Could you clarify this comment, please? I'm not sure what it's trying to suggest.

Reply:
Thanks for the replies. I will fix all the problems today.  We know that a dictionary, a list and an argument list is one logical line no matter how long it is with its entries all output in separate lines. Here it means that only when the comment is not inside this kind of logical line, aka, the comment has parent level equal to 0, we set the spaces_required_before to 0. Because we don't want the comments inside a dictionary to always align with its parent (the dictionary variable name). Examples are given in pull request README.